### DISCUSSION TITLE
Workaround for image notifications on low power/no protect plan devices

### DISCUSSION THREAD

**Discussion by thefunkygibbon** [tsightler/ring-mqtt#772] [General Discussion]
Hi, 

So after setting up ring-mqtt and trying to get it to work, running into the issue with the snapshot being sent on notifications being the same snapshot as always, and reading all the FAQs saying it isn't possible without having a protect plan:
After a bit of messing around with my automation, I've managed to get it to work and it works a treat.

The trick is to completely ignore the entity for snapshots.
What I ended up doing is to creating a Generic Camera to the RSTP URL for Ring-mqtt for each ring doorbell (i have 2). 
In the automation I have it set to create the snapshot using camera.snapshot to the relevant doorbell generic camera entity and then send the notification to my mobile app, followed immediately by a camera.record action to record what's been captured for 30 seconds.  

The result is exactly what i set out to achieve, pretty much the functionality of the protect plan without having to pay for the protect plan.  Smart notifications with an image of the motion, plus recording the activity within HA. 

Now I just need to work out if there is a better way to view and manage the media files within HA since the inbuilt media browser is frankly awful. . 

Happy to post my automation yaml if anyone is interested.

**Comment by tsightler** [comment]
It's always been possible to use an automation to grab a screenshot, or short clip, from the camera stream, regardless of subscription status, I've always been very clear about this.  Back in the day this was even implemented directly in ring-mqtt.  However, this approach is not always 100% reliable and sometimes can be subject to significant delays (3-7 seconds) before the notification is sent.  Most users didn't understand these limitations and constantly complained about the issue and these complaints were 99% from users who had crappy low-power Ring cameras and didn't pay for Ring services so I decided to simply not support this scenario and removed this support from the code once we managed to work out push notifications.

This of course doesn't mean it's not possible to create your own automation, as you have done, and I've always offered this up as an option.  However, I will point out that I have consistently made it clear that this project is not about replacing Ring paid functionality.  While this solution might be good enough for you, based on the complaints I received when this functionality was available directly in ring-mqtt, it was not good enough for most, and several took their frustrations with Ring out on me, which is why I decided to stop supporting this completely.  I would suggest you share your automation in the HA forums or on your own Github or whatever, but not here.

**Comment by thefunkygibbon** [comment]
oh right.....ok.  
i didn't find any mention of doing this, only mention that it "just doesn't work without a ring plan". 
hence spending a while messing around trying to get it done and thinking I'd accomplished something to help others. 

Using this INSTEAD of paying ring for a subscription is the main use case for me and no doubt many others given how Ring have gimped the product completely if you don't have one. 

thanks for the work on the project.

**Reply by tsightler** [reply]
> Using this INSTEAD of paying ring for a subscription is the main use case for me and no doubt many others given how Ring have gimped the product completely if you don't have one.

But it is absolutely **_NOT_** the goal of this project, and if it becomes a common use case, it's likely the Ring would block this project and other open source projects similar to what Chamberlain did with MyQ.

**Reply by thefunkygibbon** [reply]
you mean ring would block api access in general, they couldn't just block this specific project due to the way it works.   either way, if they're petty enough to cut off their nose to spite their face over the maybe 0.1% of Ring users who have the knowledge and inclination do do this and not pay them their subscription fee's, then, well, i'm sure they'll lose even more customers. 

take care

**Reply by tsightler** [reply]
MyQ is a similar example, yes they blocked it for 0.1% of users, but now all users have no access.  There's nothing that compels these companies to allow 3rd-party clients, they just happen to work because the API doesn't enforce anything unique for them, but any company can choose to block 3rd-party clients trivially at any point.  They likely wouldn't claim it was due to free users or anything like that, they would likely claim it was for "security", but regardless, it would impact everyone negatively.

I doubt in the end Ring cares about this, users of such integrations already represent a small number of their total users, and users that don't pay for subscriptions are just a cost to them, so there's little reason to care if they are lost.

I've simply stated from the start of this project that the goal is to allow customers to extend Ring products to their home automation platforms, not to bypass Ring subscriptions, and that stance will remain as long as I'm running this project.   Due to the fact that this is a violation of this projects goals, I'm going to lock this thread now.

I recognize that there are users that don't want to pay for Ring, and that's fine and fair, I just don't support those users with this project, although I of course can't stop people from using it for that purpose and you are free to share you automations anywhere else you like.
